app/src/main/java/com/connfa/service/firebase/injection/FirebaseModule.java

```diff
@@ -0,0 +1,24 @@
+package com.connfa.service.firebase.injection;
+
+import com.connfa.injection.ApplicationLifecycle;
+import com.connfa.service.firebase.FirebaseConnfaRepository;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import dagger.Module;
+import dagger.Provides;
+
+@Module
+public class FirebaseModule {
+
+    @Provides
+    DatabaseReference databaseReference() {
+        return FirebaseDatabase.getInstance().getReference();
+    }
+
+    @ApplicationLifecycle
+    @Provides
+    FirebaseConnfaRepository firebaseConnfaRepository(DatabaseReference database) {
```

Comment:
I thought of extracting an interface, sacrifying 3 rabbits to the moon and having a qualifier, then I got an arrow in the knee.